docs/examples/index.md

```diff
@@ -1384,3 +1421,123 @@ To parse the logs, one easy way is to use [jq](https://stedolan.github.io/jq/) -
+
+
+
+# <a name="example-8"></a> Example 8
+[//]: # "-------------------------------------------------------------------------------------"
+
+## General information
+
+This example covers a simple use-case of running NLP applications as one of the components of the data processing pipeline. The example application is using [GATE](https://gate.ac.uk/) suite as the NLP engine. The database schema used in this example is based on [Example 2](#example-2).
+
+[//]: # "<span style='color:red'> NOTE: </span>"
+**Note: For the moment, the example application is based on the GATE version < 8.5. The GATE in version >= 8.5 introduced a significant change in resources handling and as CogStack uses the previous version of GATE library it needs to be updated in the upcoming release.**
+
+
+## Example GATE application
+
+CogStack pipeline allows to include custom user GATE applications as one of the data processing components. These applications can be previously designed in GATE suite (e.g., using GATE Developer GUI application) and stored as a custom *gapp* or *xgapp* application with the used resources. CogStack implements *GateService* which uses GATE API to run these applications using the GATE Embedded version.
+
+
+### GATE ANNIE Gazetteer
+
+In this example, we developed a simple GATE application to annotate common drugs and medications. The application is using the GATE ANNIE plugin (with default configuration), implementing a custom version of [ANNIE Gazetteer](https://gate.ac.uk/sale/tao/splitch13.html). The application has been created in GATE Developer studio and exported into *gapp* format. This application is hence ready to be used by GATE and is stored in `gate/app` directory as `drug.gapp` alongside the used resources.
+
+The list of drugs and medications to annotate is based on a publicly available list of FDA-approved drugs and active ingredients. The data can be downloaded directly from [Drugs@FDA database](https://www.fda.gov/drugs/informationondrugs/ucm079750.htm). The list of used drugs is stored as `drug.lst` and active ingredients as `active.lst`.
+
+More information about creating custom GATE applications can be found in the official [GATE documentation](https://gate.ac.uk/documentation.html).
+
+### Data preparation
+
+This example is a basic one as our NLP application is not fully context aware and is looking in the text only for the words found in the provided list of drugs. Moreover, as the list of drugs and active ingredients can contain full names of drugs (or ingredients), the full names of drugs may not be found in the parsed text. Hence, the FDA-approved list of drugs and components needs to be further post-processed and filtered. All this is being performed by script `extra/clean_list.py` in the example directory. It uses as an input the raw data downloaded from Drugs@FDA database and a list of words to filter out (e.g., the most frequent 10000 words in English) to prepare a curated list of drugs and active ingredients.
```

Comment:
might be worth noting here how this simple dictionary match could later be expanded to use an ontology which would allow taxonomy & synonym aggregation

Reply:
Added a mention about it and that it could be possibly implemented using different GATE plugins.